### PR TITLE
first draft

### PR DESCRIPTION
**Pull request by scottcreynolds**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,26 +9,338 @@ JavaScript lodash Templates
...
+  //insert comment into "comments" div in this format:
+  //<div class="comment"><p>comment</p><p>Posted By: <span class="commenter">commenter</span></p></div>
+}
+```
+
+We know the format of the template we want, but instead of adding it
+using string concatenation, we're going to use the lodash `template`
+function.
+
+**Top-tip:** By default, we access lodash functions through the
```
Comment (pletcher) [inline README.md:91]: 👍

File: README.md [hunk 2]
```diff
@@ -9,26 +9,338 @@ JavaScript lodash Templates
...
+
+```js
+function postComment() {
+  var commenter = document.getElementById("commenterName").value;
+  var comment = document.getElementById("commentText").value;
+  //insert comment into "comments" div in this format:
+  //<div class="comment"><p>comment</p><p>Posted By: <span class="commenter">commenter</span></p></div>
+
+  //create template string
+  var commentTemplate = '<div class="comment"><p><%= comment %></p><p>Posted By: <span class="commenter"><%= commenter %></span></p></div>';
```
Comment (pletcher) [inline README.md:160]: I really like how you lead students from strings to proper templates!

File: README.md [hunk 3]
```diff
@@ -9,26 +9,338 @@ JavaScript lodash Templates
...
+
+### HTML Templates in JavaScript
+
+Putting our templates in a string isn't great for a couple of reasons.
+We don't get the human-readable, indented structure of regular HTML. We
+don't get the syntax highlighting on the tags that our editor provides.
+Adding or changing HTML inside a string is difficult and prone to error.
+What we need is a way to use a regular chunk of HTML as our template.
+
+Fortunately, lodash gives us a great way to do this. We can use a
```
Comment (pletcher) [inline README.md:189]: 👍

File: README.md [hunk 4]
```diff
@@ -9,26 +9,338 @@ JavaScript lodash Templates
...
+  var commenter = document.getElementById("commenterName").value;
+  var comment = document.getElementById("commentText").value;
+
+  //insert comment into "comments" div in this format:
+  //<div class="comment"><p>comment</p><p>Posted By: <span class="commenter">commenter</span></p></div>
+
+  //create template string
+  //var commentTemplate = '<div class="comment"><p><%= comment %></p><p>Posted By: <span class="commenter"><%= commenter %></span></p></div>';
+
+  // THIS IS THE ONLY LINE WE HAVE TO CHANGE!
```
Comment (pletcher) [inline README.md]: :tada:

File: README.md [hunk 5]
```diff
@@ -9,26 +9,338 @@ JavaScript lodash Templates
...
+```js
+// index.js
+function postComment() {
+  var commenter = document.getElementById("commenterName").value;
+  var comment = document.getElementById("commentText").value;
+
+  //insert comment into "comments" div in this format:
+  //<div class="comment"><p>comment</p><p>Posted By: <span class="commenter">commenter</span></p></div>
+
+  //create template string
```
Comment (pletcher) [inline README.md]: Do you think we should keep these lines commented out for students to reference? Or should we just have a comment about all the stuff we could remove?

Comment (scottcreynolds) [inline README.md]: yeah I went back and forth and ultimately thought it was more illustrative to comment the old `commentTemplate` line, but I could cut it too.

File: README.md [hunk 6]
```diff
@@ -9,26 +9,338 @@ JavaScript lodash Templates
...
+#### Escaping HTML Data
+
+Let's say we don't want to allow our commenters to use HTML markup in their
+comments. Right now, if you tried posting a comment like
+`<p>hi</p><p>everyone</p><h1>this is my comment</h1>`, our form would
+let you. But of course, like everything else, trolls ruin this and find
+ways to break our blog style with their markup, so we want to change our
+comment field to not execute HTML tags.
+
+Luckily, it's as simple as changing the delimiter from *interpolate* (`<%= %>`) to *escape* (`<%-
```
Comment (pletcher) [inline README.md:284]: 💯

File: index.html [hunk 7]
```diff
@@ -2,10 +2,16 @@
 <html>
   <head>
     <title>JavaScript lodash Templates</title>
+    <script src="lodash.min.js"></script>
```
Comment (pletcher) [inline index.html]: Nit: could we move this to the bottom of `<body>` (above `<script src="index.js"></script>`)?

**Comment by pletcher** [comment]
This is great! I left two minor comments, but otherwise it's just about ready!

File: README.md (hunk 5 above)
Comment (pletcher) [inline README.md]: Cool -- totally happy to leave them in :)

**Comment by scottcreynolds** [comment]
@pletcher updated

**Pull request merged**